bundles/org.dataflowanalysis.converter/src/org/dataflowanalysis/converter/DataFlowDiagramConverter.java

```diff
@@ -263,27 +265,44 @@ private void createNodes(DataFlowDiagram dataFlowDiagram, List<Child> children,
     }
 
     private void createFlows(DataFlowDiagram dataFlowDiagram, List<Child> children) {
+    	var controlFlowNameMap = createControlFlowNameMap(dataFlowDiagram);
         for (Flow flow : dataFlowDiagram.getFlows()) {
-        	fillPinToFlowNamesMap(inputPinToFlowNamesMap,flow);
-            children.add(createFlow(flow));
+        	fillPinToFlowNamesMap(inputPinToFlowNamesMap,flow, controlFlowNameMap);
+            children.add(createFlow(flow, controlFlowNameMap));
         }
     }
+    
+    private HashMap<Flow, String> createControlFlowNameMap(DataFlowDiagram dataFlowDiagram) {
+    	var controlFlowNameMap = new HashMap<Flow, String>();
+    	dataFlowDiagram.getNodes().forEach(node -> {
+    		var controlFlows = dataFlowDiagram.getFlows().stream()
+	    		.filter(flow -> flow.getDestinationNode().equals(node))
+	    		.filter(flow -> flow.getEntityName().equals("")).toList();
+    		
+    		String controlFlowName = "" + CONTROL_FLOW_NAME;
```

Comment:
Didnt you do this code in a previous PR

Reply:
Yes, to run some tests i also copied it over to  this branch